tools/validator-client-scripts/src/main.rs

```diff
@@ -0,0 +1,226 @@
+        Command::VestingClaimDelegatorRewards(args) => {
+            vesting_claim_delegator_reward(client, args).await
+        }
+        Command::VestingClaimOperatorRewards(_args) => vesting_claim_operator_reward(client).await,
+        Command::VestingCompoundDelegatorRewards(args) => {
+            vesting_compound_delegator_reward(client, args).await
+        }
+        Command::VestingCompoundOperatorRewards(args) => {
+            vesting_compound_operator_reward(client, args).await
+        }
+        Command::VestingCreateSchedule(args) => vesting_create_schedule(client, args, &denom).await,
+        Command::VestingDelegateToMixnode(args) => {
+            vesting_delegate_to_mixnode(client, args, &denom).await
+        }
+        Command::VestingUnbondGateway(_args) => vesting_unbond_gateway(client).await,
+        Command::VestingUnbondMixnode(_args) => vesting_unbond_mixnode(client).await,
+        Command::VestingUndelegateFromMixnode(args) => {
+            vesting_undelegate_from_mixnode(client, args).await
+        }
+        _ => unreachable!(),
+    }
+}
+
+fn setup_logging() {
+    let mut log_builder = pretty_env_logger::formatted_timed_builder();
+    if let Ok(s) = ::std::env::var("RUST_LOG") {
+        log_builder.parse_filters(&s);
+    } else {
+        // default to 'Info'
+        log_builder.filter(None, log::LevelFilter::Info);
+    }
+
+    log_builder
+        .filter_module("hyper", log::LevelFilter::Warn)
+        .filter_module("tokio_reactor", log::LevelFilter::Warn)
+        .filter_module("reqwest", log::LevelFilter::Warn)
+        .filter_module("mio", log::LevelFilter::Warn)
+        .filter_module("want", log::LevelFilter::Warn)
+        .filter_module("sled", log::LevelFilter::Warn)
+        .filter_module("tungstenite", log::LevelFilter::Warn)
+        .filter_module("tokio_tungstenite", log::LevelFilter::Warn)
+        .init();
+}
+
+fn create_client(args: &Args) -> Client {
+    let network_details = NymNetworkDetails::new_from_env();
+    let client_config = nymd::Config::try_from_nym_network_details(&network_details)
+        .expect("failed to construct valid validator client config with the provided network");
+    NymdClient::connect_with_mnemonic(
+        client_config,
+        args.nymd_url
+            .as_ref()
+            .expect("nymd url was not provided")
+            .as_str(),
+        args.mnemonic
+            .as_ref()
+            .expect("mnemonic was not provided")
+            .clone(),
+        None,
+    )
+    .expect("failed to create the client")
+}
+
+async fn wait_for_interrupt() {
+    if let Err(e) = tokio::signal::ctrl_c().await {
+        error!(
+            "There was an error while capturing SIGINT - {:?}. We will terminate regardless",
+            e
+        );
+    }
+    println!(
+        "Received SIGINT - the gateway will terminate now (threads are not yet nicely stopped, if you see stack traces that's alright)."
```

Comment:
gateway will terminate? : )
though it was most likely caused by me, but hey, `git blame` will show your name here hehe

Reply:
Updated